doc/userguide/rules/ldap-keywords.rst

```diff
@@ -163,3 +163,155 @@ Example of a signature that would alert if a packet has more than 2 LDAP respons
+``ldap.responses[].add_response.result_code``
+``ldap.responses[].del_response.result_code``
+``ldap.responses[].mod_dn_response.result_code``
+``ldap.responses[].compare_response.result_code``
+``ldap.responses[].extended_response.result_code``
+
+.. table:: **Result code values for ldap.responses.result_code keyword**
+
+    =========  ================================================
+    Code       Name
+    =========  ================================================
+    0          success
+    1          operations_error
+    2          protocol_error
+    3          time_limit_exceeded
+    4          size_limit_exceeded
+    5          compare_false
+    6          compare_true
+    7          auth_method_not_supported
+    8          stronger_auth_required
+    10         referral
+    11         admin_limit_exceeded
+    12         unavailable_critical_extension
+    13         confidentiality_required
+    14         sasl_bind_in_progress
+    16         no_such_attribute
+    17         undefined_attribute_type
+    18         inappropriate_matching
+    19         constraint_violation
+    20         attribute_or_value_exists
+    21         invalid_attribute_syntax
+    32         no_such_object
+    33         alias_problem
+    34         invalid_dns_syntax
+    35         is_leaf
+    36         alias_dereferencing_problem
+    48         inappropriate_authentication
+    49         invalid_credentials
+    50         insufficient_access_rights
+    51         busy
+    52         unavailable
+    53         unwilling_to_perform
+    54         loop_detect
+    60         sort_control_missing
+    61         offset_range_error
+    64         naming_violation
+    65         object_class_violation
+    66         not_allowed_on_non_leaf
+    67         not_allowed_on_rdn
+    68         entry_already_exists
+    69         object_class_mods_prohibited
+    70         results_too_large
+    71         affects_multiple_dsas
+    76         control_error
+    80         other
+    81         server_down
+    82         local_error
+    83         encoding_error
+    84         decoding_error
+    85         timeout
+    86         auth_unknown
+    87         filter_error
+    88         user_canceled
+    89         param_error
+    90         no_memory
+    91         connect_error
+    92         not_supported
+    93         control_not_found
+    94         no_results_returned
+    95         more_results_to_return
+    96         client_loop
+    97         referral_limit_exceeded
+    100        invalid_response
+    101        ambiguous_response
+    112        tls_not_supported
+    113        intermediate_response
+    114        unknown_type
+    118        canceled
+    119        no_such_operation
+    120        too_late
+    121        cannot_cancel
+    122        assertion_failed
+    123        authorization_denied
+    4096       e_sync_refresh_required
+    16654      no_operation
```

Comment:
Where do these last values come from ? Do you have some RFC references ?

Reply:
I used the [``fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result``](https://github.com/OISF/suricata/pull/12583/commits/530e9b56c93f515189981d903e289133b33e44b8#diff-68d62050879d651f59a4db94da80cc7e207398df553b216748d7e25b3f6447d5L46) function, which was removed from ``rust/src/ldap/types.rs``, as a reference.

But those values can be found at https://ldap.com/ldap-result-code-reference/

Reply:
Thanks for this URL
Can you put it in the doc ?